src/application/components/Explorer/postMessageHelpers.ts

```diff
@@ -0,0 +1,158 @@
+import type { GraphQLError, IntrospectionQuery } from "graphql";
+
+export type JSONPrimitive = boolean | null | string | number;
+export type JSONObject = { [key in string]?: JSONValue };
+export type JSONValue = JSONPrimitive | JSONValue[] | JSONObject;
+
+export const EXPLORER_LISTENING_FOR_SCHEMA = "ExplorerListeningForSchema";
+export const EXPLORER_LISTENING_FOR_STATE = "ExplorerListeningForState";
+export const EXPLORER_REQUEST = "ExplorerRequest";
+export const EXPLORER_RESPONSE = "ExplorerResponse";
+export const EXPLORER_SUBSCRIPTION_REQUEST = "ExplorerSubscriptionRequest";
+export const EXPLORER_SUBSCRIPTION_RESPONSE = "ExplorerSubscriptionResponse";
+export const SET_OPERATION = "SetOperation";
+export const SCHEMA_ERROR = "SchemaError";
+export const SCHEMA_RESPONSE = "SchemaResponse";
+export const DEV_TOOLS_ASK_FOR_GRAPH_REF_AND_AUTHENTICATE =
+  "DevTools_AskForGraphRefAndAuthenticate";
+
+export const EMBEDDABLE_EXPLORER_URL = "https://embed.apollo.local:3000";
+// "https://explorer.embed.apollographql.com";
+export const EXPLORER_SUBSCRIPTION_TERMINATION =
+  "ExplorerSubscriptionTermination";
+
+export type ExplorerResponse = {
+  data?: JSONValue | undefined;
+  errors?: readonly GraphQLError[] | undefined;
+  error?:
+    | {
+        message: string;
+        stack?: string;
+      }
+    | undefined;
+  status?: number;
+  headers?: Record<string, string>;
+};
+
+export type OutgoingMessageEvent =
+  | {
+      name: "SchemaError";
+      error?: string;
+      errors?: readonly GraphQLError[] | undefined;
+    }
+  | {
+      name: "SchemaResponse";
+      schema: IntrospectionQuery | string;
+    }
+  | {
+      name: "HandshakeResponse";
+      graphRef?: string;
+      inviteToken?: string;
+      accountId?: string;
+    }
+  | {
+      name: "ExplorerResponse";
+      operationId: string;
+      response: ExplorerResponse;
+    }
+  | {
+      name: "SetOperation";
+      operation: string;
+      variables: string;
+    }
+  | {
+      name: "ExplorerSubscriptionResponse";
+      operationId: string;
+      response: ExplorerResponse;
+    }
+  | {
+      name: "StudioUserTokenForEmbed";
+      id: string;
+      token: string;
+      graphRef: string;
+    }
+  | {
+      name: "PartialAuthenticationTokenResponse";
+      partialToken: string;
+    }
+  | {
+      name: "DevTools_AskForGraphRefAndAuthenticate";
+    };
+
```

Comment:
added typings for incoming and outgoing messages in this file - before we were just winging it